toolsrc/src/vcpkg/dependencies.cpp

```diff
@@ -1040,7 +1058,7 @@ namespace vcpkg::Dependencies
         if (has_non_user_requested_packages)
             System::print2("Additional packages (*) will be modified to complete this operation.\n");
 
-        if (!remove_plans.empty() && !is_recursive)
+        if ((!remove_plans.empty() || !rebuilt_plans.empty()) && !is_recursive)
```

Comment:
```suggestion
        if (!(remove_plans.empty() && rebuilt_plans.empty()) && !is_recursive)
```

You might want to demorgan this so that it remains "if anything to do and not recursive".